### ISSUE TITLE
Option To toggle default page buttons

### ISSUE THREAD

**Issue by McHauge** [bitfocus/companion#880]
**PLEASE USE THIS PAGE IF YOU WANT SUPPORT FOR A NEW DEVICE**

https://github.com/bitfocus/companion-module-requests/issues/new

**Describe the feature**
The option to toggle if you want all pages to have the 3 default page buttons or to keep them empty just like it is right now, where you have to add them yourself.

**Usecases**
To avoid ending up on pages where you get stuck, like how it is at the moment where if you forget to delete the button to the next page you will end up on a dead-end there you can't get home from the stream deck itself.
Or maybe just have them on as default (from the start), then the user can clear the page via that button or remove those he/she doesn't need.

**Comment by dnmeid** [comment]
Personally I like it the way it is now. Noone ever cycles thru all 99 pages. Most people use far less pages and with the new internal goto page action random access to pages is getting more and more common. As a step towards the upcoming window system without pages I would vote to keep it like it is and gradually fade the legacy page button locations.

**Comment by McHauge** [comment]
@dnmeid for me it's not about using all pages, but about not getting stuck on an empty page. had it happen 4 or 5 times while programming something on the fly for the producer at an event this weekend, whereby not actively removing the last "up" or "down" arrow ended up on an empty page with no way home. might just be because of being in a rush, but have a feeling this will become a common thing when V2.0 rolls out. but maybe not.

**Comment by dnmeid** [comment]
That is a legit point. I usually program my pages carefully, but I remember I also came to that dead end some times.
So at the moment I see the options to:
- always provide the legacy page buttons and let the user remove them on demand
- never provide the buttons and promote active programming (sometimes slower, but more real estate)
- make the toggle on/off (what happens to the buttons on "used" pages when toggeling off?)

Another two possibilities to discuss:
- Make a button to generate all the legacy pagebuttons when their position is empty
- Give an easy opportunity to switch pages on devices from the webapp

**Comment by McHauge** [comment]
Like those idea's just reposting my comment from slack not directly associated with your comment but still: 

> from a clean install all pages are blank, and if you clear the page with the page buttons, you now have two ways to end in a dead-end, when you add page two, you still got two dead ends if you don't actively remove the last up and the first down button, therefore at least from my point of view a clean install or a clear set up should at least have the option to start with all page buttons on all 99 pages, that you can then remove if you don't want to end up on those pages, that would at least prevent dead ends from a fresh start. Or at least as an option when you reset to reset with or without page buttons.
Again just my opinion, normally when I/we use companion we use different setup nearly all the time and add and remove pages even while on-air, so in 1 hour we ended stuck up on a blank page 4-5 times because we forgot to remove the last up and first down while editing pages live.

- One idea I had for a "simple" interface fix would be to have the "reset/clear all" have an option to "clear all but keep/create page buttons" or/and the option like you suggest to add page buttons to all pages where the legacy buttons are free.
- But yes the idea of promoting more active page programming makes sense, but not all setups will take the best use of this "kinda folder" based setup.
- And yes the option to quickly set a device to a page, kinda like the slider under devices already does. but maybe with better management or at least the option to just type the page you want.

**Comment by istnv** [comment]
Maybe some type of a 'shift' key or a long press on a empty button, then overlay the current page with temporary up/down buttons to move away from a blank page. Revert to normal after a few seconds.

**Comment by McHauge** [comment]
@istnv not sure how that would be done in practice, but I like the idea. on an empty page hold all four corners for 3-5sec? or something like that got no idea how to implement that though.

**Comment by Julusian** [comment]
Once #2003 is done, and it is possible to add and remove pages, I think this makes more sense.
It can be done as a setting to affect how new pages are created.
I don't think it is safe to do anything to modify existing pages

**Comment by Julusian** [comment]
I have merged in another issue into this, which is asking for a configurable layout of these buttons.  
With the dynamic page count, I think this is more viable to do now if this is something we want to do.

I am envisioning it to be a setting which affects only new pages.  
When creating a fresh config/full reset, it should not be adding the nav buttons to the single page it creates.
And it should be added to the wizard. Upon completing the wizard, if there is 1 page, and 0 controls on it, then add the chosen nav buttons.

But I wonder if this is something we actually want to do?

I don't think we should do anything fancy to avoid getting stuck on a page, anything like that would itself need to be possible to disable and might even need to be opt-in (defeating the point of it) as it would allow people configs where a streamdeck is locked to a dedicated page to escape that page
